File: BusinessLogic.cs
using System;
using System.ComponentModel.DataAnnotations;

public class ValidationHelper
{
    public static bool IsValidEmail(string email)
    {
        return new EmailAddressAttribute().IsValid(email);
    }

    public static decimal GetPriceWithDiscount(string price, decimal discount)
    {
        if (decimal.TryParse(price, out decimal result))
        {
            return result * (1 - discount);
        }
        else
        {
            throw new ArgumentException("Invalid price format.");
        }
    }
}

File: business_logic.py
class BusinessLogic:
    def calculate_price(self, price, discount):
        if discount < 0 or discount > 100:
            raise ValueError('Discount must be between 0 and 100!')
        return price * (1 - discount / 100)

    def validate_email(self, email):
        if '@' not in email or '.' not in email:
            return False
        return True
